Vagrantfile

```diff
@@ -48,11 +48,21 @@ Vagrant.configure('2') do |config|
       # meminfo returns kilobytes, convert to MB
       # v.memory = rep 'MemTotal' /proc/meminfo | sed -e 's/MemTotal://' -e 's/ kB//'to_i / 1024 / 8
       v.cpus = %x(nproc).to_i
+	elsif host =~ /mswin|mingw|cygwin/
+      # Windows. Tested on Win 7 SP1
+      # 80% of available free memory, 50% of available CPUs
+      v.memory = (`wmic OS get FreePhysicalMemory`.split[1].to_i / 1024 * 0.80).to_i
+	  cpus = `wmic computersystem Get NumberOfLogicalProcessors`.split[1].to_i / 2
+	  cpus < 1 ? v.cpus = 1 : v.cpus = cpus
     end
   end
 
   project_name = Dir.pwd.split('/').last
-
+  # The hostname set for the VM should only contain letters, numbers, hyphens or dots. It cannot start with a hyphen or dot.
+  # Strip any hypens and dots from the begining of the string
+  project_name.gsub!(/^[-.]+/, "")
+  # Strip any char that is not letter, number, hyphen or dot
+  project_name.gsub!(/[^0-9A-Za-z.-]+/, "")
```

Comment:
it's probably worth to switch those two `gsub`s places, since it's still possible to have invalid dir name (with `@.abc` name for example)

Reply:
Agree. Updated as per your advise.